src/language-js/embed.js

```diff
@@ -16,6 +17,38 @@ const {
   utils: { mapDoc, stripTrailingHardline }
 } = require("../doc");
 
+const cssPlaceholder = new Placeholder("prettier");
+const CSS_PROPERTY_PLACEHOLDER = new Placeholder("prettier").get(0);
+const CSS_EXTRA_SEMICOLON_MARK = new Placeholder("prettier").get(0);
+const placeholderPiecesToStringArray = pieces =>
+  pieces.map(({ isPlaceholder, string, placeholder }) =>
+    isPlaceholder ? placeholder : string
+  );
+
+const cleanCSS = string =>
+  string
+    // preserve `prettier-ignore` comments
+    .replace(/\/\*\s*prettier-ignore\s*\*\//g, "_")
+    .replace(/\/\/\s*prettier-ignore/g, "_")
+    // remove block comments
+    .replace(/\/\*[\s\S]*?\*\//g, "")
+    // remove quoted strings
+    .replace(/(['"]).*?(\1)/g, "_")
+    // remove strings in parentheses
+    .replace(/\(.*?\)/g, "_")
+    // remove inline comments
+    .replace(/\/\/.*/g, "");
+const cssIdentityRegExp = /[^$a-zA-Z\d_-]/;
```

Comment:
Lie for Sass/Less and non standard CSS-in-JS

Reply:
`cssIdentifierRegExp` would also be a better name

Reply:
Here’s "ident-token" in the CSS spec: https://www.w3.org/TR/css-syntax-3/#ident-token-diagram

Reply:
maybe I named it not good, I can explain what is this for, 

```css
a {
  before${js}after;
}
```

it will be

```css
a {
  beforePLACEHOLDERafter;
}
```

as I explained, postcss can't parse this,

I need do

```css
a {
  beforePLACEHOLDERafter;
}
```
but, I split them into ['....before', PLACEHOLDER, 'after...'], I wan't move before and after back, this RegExp is used to check what character can be moved.

I'm open to a good solution

Reply:
test cases are on top of this file 

https://github.com/prettier/prettier/blob/3bddcd07d2ca72dfa06f77105e2162a141774ec1/tests/template_literals/css-orphan.js

Reply:
Let's rename variable to avoid misleading